src/store/rescan.js

```diff
@@ -1,33 +1,85 @@
 import api from "@/api";
+import { fetchAll } from "./actions";
 
 export default {
   namespaced: true,
   state: {
-    rescan: null,
+    rescans: {},
     lastClick: new Date(0),
     loading: false,
   },
   mutations: {
+    setRescans(state, payload) {
+      const oldRescans = state.rescans;
+      state.rescans = {};
+      for (let id in oldRescans) {
+        state.rescans[id] = oldRescans[id];
+      }
+      const loaded = new Date();
+      for (let rescan of payload) {
+        rescan.loaded = loaded;
+        state.rescans[rescan.id] = rescan;
+      }
+    },
+    setRescan(state, { id, rescan }) {
+      const oldRescans = state.rescans;
+      state.rescans = {};
+      for (let id in oldRescans) {
+        state.rescans[id] = oldRescans[id];
+      }
+      rescan.loaded = new Date();
+      state.rescans[id] = rescan;
+    },
+    setStartLoading(state) {
+      state.startLoading = new Date();
+    },
+    removeOld(state) {
+      const oldRescans = state.rescans;
+      state.rescans = {};
+      for (let id in oldRescans) {
+        if (oldRescans[id].loaded > state.startLoading) {
+          state.rescans[id] = oldRescans[id];
+        }
+      }
+    },
     setLastClick(state, payload) {
       state.lastClick = payload;
     },
     setLoading(state, payload) {
       state.loading = payload;
     },
-    setRescan(state, payload) {
-      state.rescan = payload;
-    },
   },
   actions: {
-    async show({ commit, rootState }) {
+    async index({ commit, rootState, getters }) {
+      if (rootState.rescan.loading) {
+        return true;
+      }
+      try {
+        commit("setLoading", true);
+        do {
+          const generator = api.rescans.index(rootState.auth);
```

Comment:
Did you test this? It crashes here for me because the module is still called `rescan` in `api-client-js` instead of `rescans`.

Reply:
I'd made this change to `api-client-js` but hadn't pushed it yet 🤦